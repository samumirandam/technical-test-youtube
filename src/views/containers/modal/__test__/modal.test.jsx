/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-unresolved */
import React from 'react';
import { render, screen, fireEvent } from '@utils/test-utils';

import Modal from '../index';

const defaultProps = {};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(
    <Modal {...setupStore}>
      <p>modal test</p>
    </Modal>,
    {},
  );
};

describe('Test for Modal component', () => {
  test('Should render without errors', () => {
    setup();
    expect(screen.getByText('modal test')).toBeTruthy();
  });

  test('Should renders the same component', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test('Should render click in button', () => {
    const props = {
      primaryButton: 'test button',
    };
    setup(props);
    fireEvent.click(screen.getByText('test button'));
    fireEvent.click(screen.getByTestId('close'));
    expect(screen.getByText('modal test')).toBeTruthy();
  });
});
