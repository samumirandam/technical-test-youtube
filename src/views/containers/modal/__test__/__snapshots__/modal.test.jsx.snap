// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Modal component Should renders the same component 1`] = `
<div>
  <div
    class="Modal"
  >
    <div
      class="Modal__container"
    >
      <div
        class="Modal__header"
      >
        <button
          class="Button Modal__header-button flat"
          data-testid="close"
          type="button"
        >
          <svg
            class="Modal__header-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 352 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            />
          </svg>
        </button>
      </div>
      <div
        class="Modal__body"
      >
        <p>
          modal test
        </p>
      </div>
      <div
        class="Modal__footer"
      />
    </div>
  </div>
</div>
`;
